File: lab/lab_1/python/lab1.py
def read_file(filename):
    try:
        with open(filename, 'r') as file:
            content = file.read()

            if not content:
                print("The file is empty.")
            else:
                print(content, end='')

    except FileNotFoundError:
        print(f"Error: File not found : {filename}")
    except IOError as e:
        print(f"Error reading file: {e}")


def main():
    filename = input("Enter filename: ")
    read_file(filename)


if __name__ == "__main__":
    main()

File: lab/lab_1/java/lab1.java

import java.io.*;
import java.util.Scanner;

public class lab1 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter filename: ");
        String fileName = scanner.nextLine();

        try (BufferedReader reader = new BufferedReader(new FileReader(fileName))) {

            String line;
            boolean isEmpty = true;

            while ((line = reader.readLine()) != null) {
                System.out.println(line);
                isEmpty = false;
            }

            if (isEmpty) {
                System.out.println("The file is empty.");
            }

        } catch (FileNotFoundException e) {
            System.out.println("Error: File not found : " + fileName);
        } catch (IOException e) {
            System.out.println("Error reading file: " + e.getMessage());
        }

        scanner.close();

    }
}
